Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,215 @@
+    ///
+    /// Then let (a+2*(b+c)+d)/6 be NUM and (e+2*(f+g)+h)/6 be DENOM.
+    /// <para>RVI = SMA(NUM)/SMA(DENOM)</para>
+    /// for a specified period.
+    /// 
+    /// https://www.investopedia.com/terms/r/relative_vigor_index.asp
+    /// </summary>
+    public class RelativeVigorIndex : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly RollingWindow<IBaseDataBar> _previousInputs;
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class.
+        /// </summary>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(int period, MovingAverageType type)
+            : this($"RVI({period},{type})", period, type)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class.
+        /// </summary>
+        /// <param name="name">The name of this indicator.</param>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(string name, int period, MovingAverageType type = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = period + 3;
+            CloseBand = type.AsIndicator("_closingBand", period);
+            RangeBand = type.AsIndicator("_rangeBand", period);
+            _previousInputs = new RollingWindow<IBaseDataBar>(3);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class along with a signal term.
+        /// </summary>
+        /// <param name="name">The name of this indicator.</param>
+        /// <param name="signalName">The name of the signal associated with this indicator.</param>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(
+            string name, string signalName, int period,
+            MovingAverageType type = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = period + 3;
+            CloseBand = type.AsIndicator("_closingBand", period);
+            RangeBand = type.AsIndicator("_rangeBand", period);
+            _previousInputs = new RollingWindow<IBaseDataBar>(3);
+            Signal = new RviSignal(signalName);
+        }
+
+        /// <summary>
+        /// An optional signal line which behaves like a slowed version of the RVI.
+        /// </summary>
+        public RviSignal Signal { get; }
+
+
+        /// <summary>
+        /// Gets the band of Closes for the RVI.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> CloseBand { get; }
```

Comment:
nit - remove extra new lines and move class members to top of class and organize 😄

Reply:
Sounds good -- will use the ordering of https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md from hereon!